lib/dolly/document.rb

```diff
@@ -143,9 +143,16 @@ def self.property *ary
       options           = ary.pop if ary.last.kind_of? Hash
       options         ||= {}
 
-      ary.each do |name|
+      if ary.count==1 && options[:class_name] == Hash && block_given?
```

Comment:
currently I've only allowed hash to have sub properties, I am considering how to extend this to properties of other classes.